### ISSUE TITLE
Address aliases support

### ISSUE THREAD

**Issue by foxcpp** [foxcpp/maddy#18]
Perhaps something like that:
```
smtp ... {
  aliases path_to_file_with_mappings
}
```
Format same as postfix's text representation (http://www.postfix.org/aliases.5.html). Except probably we will leave `|command`, `/file`, `:include` support out for now.

**Comment by foxcpp** [comment]
#### Generic solution - rewrite operation

##### Single address rewrite defined directly in configuration

```
rewrite postmaster@emersion.fr contact@emersion.fr
```

Generic form:
```
rewrite <initial address> [new address...]
```

This will replace a single address with multiple:
```
rewrite postmaster@emersion.fr contact@emersion.fr postmaster@emersion.fr
```
Rewrites are applied only once, so the directive above will not loop forever.

Regexp can be used in "Initial address" by wrapping it with /:
```
rewrite /(postmaster|admin)(@emersion\.fr)?/ contact@emersion.fr
```

##### Aliases mappings

More advanced rewriting (including mappings from files) is done using generic filter interface:
```
filter alias_map /etc/maddy/aliases 
```

We define a singleton module `alias_map` that performs rewriting using postfix's aliases file.

**Comment by emersion** [comment]
LGTM

**Comment by NamedKitten** [comment]
This is a REQUIRED feature and I NEED it to be a thing. Please don't assign this task to me. I just really require this.

**Comment by NamedKitten** [comment]
Will this work in reverse, rewriting the From field in sent emails?

**Comment by foxcpp** [comment]
No.

**Comment by NamedKitten** [comment]
Will accounts that match the new address in rewrite argument be allowed to send as the new address email?

**Comment by foxcpp** [comment]
At least, for now, we don't plan to add any checks regarding whether either From header or MAIL FROM matches actual sender account.

**Comment by Avamander** [comment]
Postfix can read these out of a database, it would be really nice if Maddy could as well.

**Comment by foxcpp** [comment]
I guess something like that could work for database-based rewrites.
```
filter alias_sql {
  driver ...
  dsn ...

  query "SELECT target FROM aliases WHERE username = %u"
}
```

A user-specified query should return a single string column containing new addresses, multiple rows are allowed, obviously. If the query fails for whatever reason (including zero rows result) - no rewrite it done.

**Comment by foxcpp** [comment]
Apparently, 7efcd81 says this:
>*Does not* close #18, that issue is for file-based mapping.

But GitHub decided to interpret this as "closes #18" and closed this issue, reopening.
